torchgeo/datasets/utils.py

```diff
@@ -819,3 +819,29 @@ def path_is_vsi(path: str) -> bool:
     .. versionadded:: 0.6
     """
     return "://" in path or path.startswith("/vsi")
+
+
+def array_to_tensor(array: np.typing.NDArray[Any]) -> Tensor:
+    """Converts a :class:`numpy.ndarray` to :class:`torch.Tensor`.
+
+    :func:`torch.from_tensor` rejects numpy types like uint16 that are not supported
+    in pytorch. This function instead casts uint16 and uint32 numpy arrays to an
+    appropriate pytorch type without loss of precision.
+
+    For example, a uint32 array becomes an int64 tensor. uint64 arrays will continue
+    to raise errors since there is no suitable torch dtype.
+
+    The returned tensor is a copy.
+
+    Args:
+        array: a :class:`numpy.ndarray`.
+
+    Returns:
+        A :class:`torch.Tensor` with the same dtype as array unless array is uint16 or
+        uint32, in which case an int32 or int64 Tensor is returned, respectively.
```

Comment:
Can you add:
```rst
.. versionadded:: 0.6
```
to the end? That will document which versions of TorchGeo support this.